### DISCUSSION TITLE
How can i install Sulu as Symfony recipe

### DISCUSSION THREAD

**Discussion by emircanerkul** [sulu/sulu#8165] [Q&A]
As a new comer, I want to use sulu as symfony recipe, but https://docs.sulu.io/en/2.6/book/getting-started.html do not show that.

In https://symfony.com/components/Symfony%20Flex I can see Sylius there but not Sulu?

So I created symfony 7 app, added bundles come with standard web skeleton then i want to add sulu, how can i? Then I may want to add api platform then Sylius.. I see your recent blog post how well sylius + sulu work together but what about api platform? or idk any other flex bundles?

As Drupal dev, https://www.drupal.org/project/ideas/issues/3327308 community getting bigger but this has a lot of cons. I rather embrace whole community rather then developing things again and again under some trademark name/logo with tons of bugs/less battle tested. Like drupal commerce, or drupal's own api frameworks.. etc. etc.

Thats why i want to install sulu as symfony flex package.

Btw weirdly i saw sulu dont support sqlite.. why? symfony.com/doc/current/doctrine.html has support? Looks those things indicates sulu not work well as bundle (of) "bundle". hopefully not.

BTW thank you for symfony 7 support.

**Comment by emircanerkul** [comment]
Tried to add sulu/sulu.

So far failed :) Could be better DX.

What i did is,

added
```
        "jackalope/jackalope-doctrine-dbal": "^1.10 || ^2.0",
        "phpcr/phpcr-migrations-bundle": "^1.6",
        "phpcr/phpcr-shell": "^1.6"
```
downgraded "symfony/*": "7.1.*" -> "symfony/*": "6.*"
because
looks sulu is not fully compatable with symfony 7 😞 

```
    - sulu/sulu 2.6.0 requires symfony/asset ^5.4 || ^6.0 -> found symfony/asset[v5.4.0, ..., v5.4.40, v6.0.0, ..., v6.4.8] but it conflicts with your root composer.json require (7.1.*).
```

But more and more errors appears.. 
```
!!  In ParameterBag.php line 93:
!!                                                                                                                                         
!!    The parameter "massive_search.adapter.zend_lucene.basepath" has a dependency on a non-existent parameter "kernel.root_dir". Did you  
!!     mean one of these: "kernel.project_dir", "kernel.build_dir", "kernel.cache_dir", "kernel.logs_dir"?                                 
!!    
```

anyway looking forward to see your comments.

**Comment by emircanerkul** [comment]
Ah okay, looks i have to use `"minimum-stability": "dev",` 

But I have 
```
 sulu/sulu 3.0.x-dev requires doctrine/orm ^2.13 -> found doctrine/orm[2.13.0, ..., 2.21.x-dev] but it conflicts with your root composer.json require (^3.3).
```
then
```
  Problem 1
    - Root composer.json requires sulu/sulu ~3 -> satisfiable by sulu/sulu[3.0.x-dev].
    - jackalope/jackalope[1.4.5, ..., 1.x-dev, 2.0.0-beta1, ..., 2.x-dev] require jackalope/jackalope-transport * -> could not be found in any version, but the following packages provide it:
      - jackalope/jackalope-doctrine-dbal Jackalope Transport library for Doctrine DBAL
      - jackalope/jackalope-jackrabbit Jackalope Transport library for Jackrabbit
      - jackalope/jackalope-fs Jackalope Transport library using the filesystem
      - jackalope/jackalope-mongodb Jackalope Transport library
      - jackalope/jackalope-prismic Jackalope Transport library for http://prismic.io DBAL
      Consider requiring one of these to satisfy the jackalope/jackalope-transport requirement.
    - sulu/sulu 3.0.x-dev requires jackalope/jackalope ^1.4.5 || ^2.0 -> satisfiable by jackalope/jackalope[1.4.5, 1.4.6, 1.4.7, 1.x-dev, 2.0.0-beta1, ..., 2.x-dev].
```
then end up
```
!!    The parameter "massive_search.adapter.zend_lucene.basepath" has a dependency on a non-existent parameter "kernel.root_dir". Did  
!!     you mean one of these: "kernel.project_dir", "kernel.build_dir", "kernel.cache_dir", "kernel.logs_dir"?                         
!!         
```

**Comment by mario-fehr** [comment]
Hi @emircanerkul 

[https://symfony.com/components/Symfony%20Flex](https://symfony.com/components/Symfony%20Flex) this link only says that these projects use Flex, but not that they can be installed as standalone Flex recipe bundles. For both Sulu and Sylius you should install the respective skelton. As described in the documentation. For sulu: [https://github.com/sulu/skeleton](https://github.com/sulu/skeleton). If you want to run both in parallel as in the blog post, I would recommend to install both skeltons in a sulu and sylius folder and then use a monorepo.

___

For installing api platform into sulu I found a WIP [here](https://github.com/sulu/sulu-demo/pull/57) but it looks like there a some issues with the core.

___

If you have any other questions or need help, don't hesitate to ask them in the[ Sulu slack.](https://sulu.io/services/support)

**Comment by alexander-schranz** [comment]
To add Sulu to existing Symfony app: Just start a new sulu skeleton and copy step by step your symfony app to it. Including your .git directory to not loos your git history of your files.

See following blog post: https://sulu.io/blog/adding-sulu-cms-to-an-existing-symfony-project

**Comment by emircanerkul** [comment]
@mario-fehr hmm when i see api-platform in there i thought they all can be used as flex package, ah okay.

I see, if the naming follows smth-bundle, which means can work together or [those](https://symfony.com/bundles) 

<img width="294" alt="image" src="https://github.com/user-attachments/assets/162258dd-14c7-4fd3-bd47-003557fd7ad8">

I also see Sulu is consist of bundles explained here: https://docs.sulu.io/en/2.6/bundles/index.html and i assume if i grab all together which should give me the same result 👶🧩🧸 but i guess there still somethings waiting to be converted into bundle.

I see yea monorepo works but there would be multiple database also quite much duplicate codebase (what if i want to install api platform both or i just want to use same generalized category bundle for both)

How much of the project is consist of bundles? I still think there should be a way to become SuluCMSBundle, could be there whole epic for this, idk could be useful and prevent becoming huge big paralyzed mono project which don't work well/adapt easily.

And indeed this is question like ticket thats why this should be in slack. But hopefully those ideas will leads the bright feature

**Comment by alexander-schranz** [comment]
@emircanerkul Sulu currently can not work as a Recipe as Symfony does not allow to override `public/index.php`, `src/Kernel.php` and symfony bundle configuration files as a recipe.

This is a known problem we @sulu and also our friends at @sylius have. We could host our own recipes repository, which I think sylius trying now but somebody would need then keep it in sync with Symfonys, which would at current state a lot of work without a lot of benefits.

More information about that you see here in the Symfony issue: https://github.com/symfony/flex/issues/957

So at current case checkout https://sulu.io/blog/adding-sulu-cms-to-an-existing-symfony-project give it a try its really not much work, that way.

**Comment by emircanerkul** [comment]
Thank you @alexander-schranz for the explanation. Yea, definitely there would be maintenance cost to manage dozens of sub repos. And if there are no workaround instead overriding kernel yea looks no other way there. Patching also a bit dirty.

Just brainstorming; if there kernel listeners, we can subscribe and run our logic, if its not enough could be sub kernel.. idk..

https://symfony.com/doc/current/components/http_kernel.html#component-http-kernel-kernel-request

kernel.request
  - sulu_kernel
  - sylius_kernel
  - etc

**Comment by emircanerkul** [comment]
Hi from era of ai agents in 2025

I have no urgent needs of this framework but still would be great if it work as symfony recipe

**Comment by mario-fehr** [comment]
@alexander-schranz sorry for bringing this up again but some how shopware got it to work to override `public/index.php`, `src/Kernel.php` and symfony bundle configuration files as a recipe. See [https://github.com/shopware/recipes](https://github.com/shopware/recipes) in detail see [https://github.com/shopware/recipes/blob/main/shopware/core/6.7/public/index.php](https://github.com/shopware/recipes/blob/main/shopware/core/6.7/public/index.php)
